teos/src/config.rs

```diff
@@ -242,6 +242,94 @@ impl Config {
     pub fn is_default(&self) -> bool {
         self == &Config::default()
     }
+
+    /// Gets a collection of custom options using vector.
+    pub fn get_non_default_options(&self) -> Vec<(String, String)> {
+        let mut non_default_options = Vec::new();
+        let default_config = Config::default();
+
+        // API
+        if self.api_bind != default_config.api_bind {
+            non_default_options.push(("api_bind".to_string(), self.api_bind.to_string()));
+        }
+        if self.api_port != default_config.api_port {
+            non_default_options.push(("api_port".to_string(), self.api_port.to_string()));
```

Comment:
We can avoid writing this long method if we can programmatically loop over the config options, this will also be more extensible.

Something like:
```rust
    /// Gets a collection of custom options using vector.
    pub fn get_non_default_options(&self) -> Vec<(String, String)> {
        let mut non_default_options = Vec::new();

        let json_default_config = serde_json::json!(&Config::default());
        let json_config = serde_json::json!(&self);

        for (key, value) in json_config.as_object().unwrap().iter() {
            if *value != json_default_config[key] {
                non_default_options.push((key.to_string(), value.to_string()));
            }
        }

        return non_default_options;
	}
```

But this means `Config` must implement `serde::Serialize`.
Also note not to leak any confidential data (like `btc_password`) to the terminal.

Reply:
lol I forgot how github works xD
thought `Add a single comment` starts a review :joy:

Reply:
> 

So if key is btc_rpc_user or btc_rpc_password I should continue?

Reply:
Oh damn, this is clever. I was wondering how to iterate over `Config` fields and never thought about turning it back to `json`.

I like this.